include/libnpy/npy.hpp

```diff
@@ -80,11 +80,12 @@ inline void read_magic(std::istream& istream, unsigned char& v_major, unsigned c
   istream.read(buf, magic_string_length+2);
 
   if(!istream) {
-    throw std::runtime_error("io error: failed reading file");
+    ARM_COMPUTE_THROW("io error: failed reading file");
```

Comment:
This is a third party library, we need to upstream the changes in a cleaner way.

Reply:
Would that be acceptable to define `ARM_COMPUTE_THROW` in this file? Or do you have something else in mind?

Reply:
Are you sure this is needed ? 
I just had a quick look and it looks like this is only included in `utils/Utils.h` and `tests/AssetsLibrary.h` ?
So I think if you build with `examples=0 benchmark_tests=0 validation_tests=0` then you should be fine.

Reply:
actually it's not needed!
My problem started with wrong assumptions that I need to include `utils/Utils.h`.
It turns out that the only change needed is that to `arm_compute/core/CL/OpenCL.h`